src/query/api/v1/middleware/metrics.go

```diff
@@ -73,6 +77,13 @@ func ResponseMetrics(opts options.MiddlewareOptions) mux.MiddlewareFunc {
 				path = "unknown"
 			}
 
+			customScopeName := r.Header.Get(headers.CustomResponseMetricsScope)
+			if customScopeName != "" {
```

Comment:
nit: can potentially generalize the case without a header to go through the same path? Not necessarily worth bothering with though